feat(schemas): design character, location, and relationship schemas for RAG system

yididiabera opened the pull request:
#### Summary
Initial schema design for Scene Consistency RAG System.

- Designed `character_schema.json` and `location_schema.json` with strong validation
- Defined `relationship_schema.json` and shared `metadata_schema.json` for entity linking and provenance
- Documented schema logic, validation patterns, and embedding integration guidelines
- Established foundation for vector DB integration

#### Scope
Schema design and documentation only. No runtime or database implementation.

#### Key Features
- **Validation:** Cloud paths, flexible LoRA triggers, underscore-friendly IDs
- **RAG-optimized:** Entity embeddings + relationship filtering for semantic retrieval
- **Unified metadata:** Consistent provenance across all entities
- **Verified integrity:** All schema and example validations pass

#### Validation Results
✅ Character schema valid  
✅ Location schema valid  
✅ Relationship schema valid  
✅ Relationships collection schema

yididiabera committed:
feat(schemas): design character, location, and relationship schemas for RAG system

tensae-b commented:
The schema is well structured and documented. great work!
**some potential imporvements**
1. Settings consistency
        Currently, tags cover general scene features (indoor, office_setting) but don’t explicitly define the full setting context.
        Consider a separate setting object with fields like:
        "setting": {
          "location": "office",
          "time_of_day": "daytime",
          "weather": "sunny",
          "props": ["desk", "laptop"]
        }
        This would make scene-level consistency easier to enforce in RAG prompts.
2. Versioning / history
For evolving characters, a version number or updated_at field helps track canonical changes

yididiabera committed:
feat(schema): add setting context and version tracking to support RAG scene consistency

yididiabera commented:
> The schema is well structured and documented. great work! **some potential imporvements**
> 
> 1. Settings consistency
>    Currently, tags cover general scene features (indoor, office_setting) but don’t explicitly define the full setting context.
>    Consider a separate setting object with fields like:
>    "setting": {
>    "location": "office",
>    "time_of_day": "daytime",
>    "weather": "sunny",
>    "props": ["desk", "laptop"]
>    }
>    This would make scene-level consistency easier to enforce in RAG prompts.
> 2. Versioning / history
>    For evolving characters, a version number or updated_at field helps track canonical changes

I've updated the schemas based on your suggestions.
- **Settings Consistency**: I've added a setting object to the location_schema to handle contextual scene metadata and cleaned up the tags arrays in both the character and location schemas to only include inherent traits.
- **Versioning**: I've added an entity_version field to both the character and location schemas to track canonical revisions.
- **Documentation**: The SCHEMA_DOCUMENTATION.md file has been updated to reflect all the v1.1 changes.
